### ISSUE TITLE
[testing] TypeError: contract.subscribe is not a function

### ISSUE THREAD

**Issue by BreatheSmile** [8xprotocol/monorepo#59]
Hi hi,

Newbie to github, sorry if I'm missing something.

I wanted to work on #49 so I tried to install the repo and run the tests. I needed to fix babel first, adding a .babelrc file and adding requires in truffle.js, as [explained here](https://ethereum.stackexchange.com/questions/21210/syntaxerror-unexpected-token-import-on-truffle-test/21409). Maybe that's a separate issue? Or did I miss something?

Anyway, running the tests gave me 9 instances of "TypeError: contract.subscribe is not a function" in the VolumeSubscription contract. (All other tests completed successfully.)  I attached the full testing output in a .txt file: [VolumeSubscription testing errors.txt](https://github.com/8xprotocol/monorepo/files/2522970/VolumeSubscription.testing.errors.txt)

At this point I'm not sure I can contribute if the existing tests are causing trouble (on my fork), so I figured I'd report this first before starting work on anything else. Cheers!

**Comment by BreatheSmile** [comment]
Thank you for the prompt pointers!

I should have mentioned I am on Windows 10 (1803), which seems to be making things a little more complicated.

Building nvm was a breeze with [nvm-windows](https://github.com/coreybutler/nvm-windows).

Lerna also installs merrily, but `lerna run build` has given me several issues. First it turned out I needed to install typescript. A simple `npm install -g typescript` fixed that. Then, I removed .js from Windows' PathExt to avoid the lerna run triggering (a misguided Windows attempt at) execution of truffle.js, [as per here](https://ethereum.stackexchange.com/questions/19347/running-truffle-migrate-just-opens-my-truffle-js-file-and-does-nothing/19444). 

I am now stuck at:

> lerna ERR! npm run build stderr:
'node_modules' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @8xprotocol/contracts@0.2.0 generate-artifacts-typings: `node_modules/@0xproject/abi-gen/bin/abi-gen.js --abis '../artifacts/src/build/abi/json/**/*.json' --out '../artifacts/src/build/wrappers' --template './contract_templates/contract.mustache' --partials './contract_templates/partials/*.mustache'`

This seems to be due to Windows handling relative paths differently from Linux ([which maybe npm is not handling well, as compared to yarn](https://github.com/yarnpkg/yarn/issues/1729)). My next step would be to just install Linux, or at least [WSL](http://wsl-guide.org/en/latest/installation.html), though I don't know if that won't open another can of worms. Any other tips? :)

**Comment by kermankohli** [comment]
Ah yeah it looks like the paths aren't resolving properly. Essentially that piece of code generates the typescript typings for the JS library to use. I'm pretty sure if you disabled it from the build script you could crack onto working on the contracts and testing should work fine. The other option is to move to yarn but that's a bit of a migration process.

Cheers,
Kerman

**Comment by BreatheSmile** [comment]
Hi hi, thanks again, I tried out some more stuff but still not seeing the end of it.

> I'm pretty sure if you disabled it from the build script

If, in packages\contracts\package.json, I take out `npm run generate-artifacts-typings` from the build script and then `lerna run build` in the root, I get errors when building the artifacts package, saying that it is missing modules in ./build/wrappers . Instead, I got `lerna run build` to complete without complaints by editing `generate-artifacts-typings` itself. I added `node` in front of the path to the .js file, and swapped out the single quotes for escaped double quotes, so that it now reads:

> `"generate-artifacts-typings": "node node_modules/@0xproject/abi-gen/bin/abi-gen.js --abis \"../artifacts/src/build/abi/json/**/*.json\" --out \"../artifacts/src/build/wrappers\" --template \"./contract_templates/contract.mustache\" --partials \"./contract_templates/partials/*.mustache\""`

After this, I can `lerna run build` without trouble, but `truffle test` still does not work. It gives me the same problem as in earlier posts; truffle.js does not have the babel requires that it seems to need. If I add

> `require('babel-register');`
> `require('babel-polyfill');`

then I can `truffle test` but I still seem to get the same 9 errors in testing VolumeSubscription that I reported in my opening post, in the .txt file.

I'm feeling a bit hopeless about installing this project at this point. I hope it's helpful to at least see what difficulties fresh-install (and fresh-faced) windows users are running into. Let me know if you think there are other meaningful things to try. Cheers!

**Comment by BreatheSmile** [comment]
Thank you Kerman!

I can get to a point where all tests pass successfully! Closing this issue as I am able to navigate the install process, which was kind of the point of opening it.

Note that I still need to do several manual fixes, all described above, to navigate installation; I have to `npm install typescript`, add the babel requires into truffle.js, and rephrase the `generate-artifacts-typings` so that it is Windows-friendly.
